### PR TITLE
chore: Update NodeJs Engine compat to v14, add pre-reqs to Composer readme

### LINKED ISSUE #8211: Build requires Node.js >=14 - but docs state otherwise
## Describe the bug
According to the [documentation](https://github.com/microsoft/BotFramework-Composer/blob/main/docs/1.3-supported-os.md) building should succeed on Node.js 12.x and 10.x.

I tried building on Node.js 12 and got this error on `yarn install`:

```
error antlr4@4.9.2: The engine "node" is incompatible with this module. Expected version ">=14". Got "12.22.1"
error Found incompatible module.
```

Apparently this module requires Node.js >= 14. Is this intentional?

## Version

Latest main branch source

## Browser

Not there yet. Still building.

## OS

- [ ] macOS
- [x] Windows
- [ ] Ubuntu

## To Reproduce

Follow the build instructions:

* git clone
* yarn install

Bam. Error.

## Expected behavior

I expect it to build. Or the documentation to state that Node.js 14.x is mandatory.

## Screenshots

image

## Additional context

Clean Azure Win 10 VM to rule out any side effects of dirty dev machines.

Note: Using Node.js 14 I could successfully install, build and launch the Composer.

### PR DESCRIPTION
**Pull request by taicchoumsft**
## Description
Update Node Version to V14.  V16 is not supported yet by bf-cli (via [orchestratorlib](https://github.com/microsoft/botframework-cli/blob/8078cc695c51725c69948bf805a3367a6e531f20/packages/orchestratorlib/package.json#L15)), and we've dropped support for Node V12 and below (via [Antlr 4.9.2](https://github.com/microsoft/botframework-cli/pull/1254/files) in bf-lu)

Also update all our docs on this site to mention we only support building Composer on NodeV14 for now.  @JonathanFingold has already done the accompanying update on the docs site [here](https://github.com/MicrosoftDocs/composer-docs-pr/pull/588)

Also add build pre-reqs to our README as a source of truth for docs. 

## Task Item
closes #8211

### CODE REVIEW

File: Composer/package.json [hunk 1]
```diff
@@ -27,7 +27,7 @@
     "ejs": "^3.1.6"
   },
   "engines": {
-    "node": ">=12"
+    "node": "14.x"
```
Comment (srinaath) [inline Composer/package.json:30]: @taicchoumsft Im assuming the build broke with Node 16, hence you are avoiding the usage of >=14?

Comment (taicchoumsft) [reply Composer/package.json:30]: Yes,   this is the error from `bf-orchestrator` when running `yarn install` on `Node V16.4.0`:

```
error @microsoft/bf-orchestrator@4.13.0-rc0: The engine "node" is incompatible with this module. Expected version "^10.0.0 || ^12.0.0 || ^13.0.0 || ^14.0.0". Got "16.4.0"
error Found incompatible module.
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
```

I just created a ticket to track this on cli side: [cli#1258](https://github.com/microsoft/botframework-cli/issues/1258)

**Review by srinaath — APPROVED** [review]

File: docs/supported-os.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+Windows 10        | Version 1607+      | x32, x64
+Windows Server    | 2012 R2+           | x32, x64
+Nano Server       | Version 1803+      | x32, x64
+
+
+## macOS
+
+OS                | Version            | Architectures
+------------------|--------------------|---------------
+Mac OS X          | 10.14+             | x64
```
Comment (hatpick) [inline docs/supported-os.md:18]: should we note something about M1 (arm) version?

Comment (taicchoumsft) [reply docs/supported-os.md:18]: Thanks @hatpick , added a sentence explicitly mentioning we don't run on M1 hardware.

**Review by hatpick — APPROVED** [review]

**Pull request merged**